Sources/CreateOptions/PackageDeclaration.swift

```diff
@@ -0,0 +1,58 @@
+import Foundation
+
+public enum DependencyRule: Decodable {
+    case exact(version: String)
+    case upToNextMajor(from: String)
+    case upToNextMinor(from: String)
+    case range(from: String, to: String)
+    case closedRange(from: String, to: String)
+    case branch(name: String)
+    case commit(hash: String)
+    case from(version: String)
+
+    var declaration: String {
+        switch self {
+        case .exact(version: let version):
+            return ".exact(\"\(version)\")"
+        case .upToNextMajor(from: let version):
+            return ".upToNextMajor(from: \"\(version)\")"
+        case .upToNextMinor(from: let version):
+            return ".upToNextMinor(from: \"\(version)\")"
+        case .range(from: let min, to: let max):
+            return "\"\(min)\"..<\"\(max)\""
+        case .closedRange(from: let min, to: let max):
+            return "\"\(min)\"...\"\(max)\""
+        case .branch(name: let name):
+            return ".branch(\"\(name)\")"
+        case .commit(hash: let hash):
+            return ".revision(\"\(hash)\")"
+        case .from(let version):
+            return "from: \"\(version)\""
+        }
+    }
+}
+
+public struct PackageDeclaration: Decodable {
+    
+    let url: URL
+    let module: String
+    var rule: DependencyRule
+    
+    public var packageDeclaration: String {
+        ".package(url: \"\(url)\", \(rule.declaration))"
+    }
+    
+    public var productDeclaration: String {
+        let cleanedPackage = url
```

Comment:
This is done per [SPM 0292](https://github.com/apple/swift-evolution/blob/main/proposals/0292-package-registry-service.md).

Reply:
I don't think its down to us to clean this though? That should be on the user right? Just like we don't validate/clean the versioning strings?

Reply:
Sorry, I missed how/why we are doing the cleaning.. This is for the package name and not the URL itself, makes perfect sense!